### PR TITLE
Drop Julia 0.5

### PR DESCRIPTION
**Pull request by omus**

### CODE REVIEW

File: src/FixedPointDecimals.jl [hunk 1]
```diff
@@ -102,23 +102,23 @@ end
 
 const FD = FixedDecimal
 
-floattype{T<:Union{Int8, UInt8, Int16, UInt16}, f}(::Type{FD{T, f}}) = Float32
-floattype{T<:Integer, f}(::Type{FD{T, f}}) = Float64
-floattype{f}(::Type{FD{BigInt, f}}) = BigFloat
+floattype(::Type{FD{T, f}}) where {T<:Union{Int8, UInt8, Int16, UInt16}, f} = Float32
+floattype(::Type{FD{T, f}}) where {T<:Integer, f} = Float64
+floattype(::Type{FD{T, f}}) where {T<:BigInt, f} = BigFloat
```
Comment (TotalVerb) [inline src/FixedPointDecimals.jl]: `BigInt` is a concrete type

File: src/FixedPointDecimals.jl [hunk 2]
```diff
@@ -102,23 +102,23 @@ end
...
+abs(x::FD{T, f}) where {T, f} = reinterpret(FD{T, f}, abs(x.i))
 
-+{T, f}(x::FD{T, f}, y::FD{T, f}) = reinterpret(FD{T, f}, x.i+y.i)
--{T, f}(x::FD{T, f}, y::FD{T, f}) = reinterpret(FD{T, f}, x.i-y.i)
++(x::FD{T, f}, y::FD{T, f}) where {T, f} = reinterpret(FD{T, f}, x.i+y.i)
+-(x::FD{T, f}, y::FD{T, f}) where {T, f} = reinterpret(FD{T, f}, x.i-y.i)
 
 # wide multiplication
-Base.@pure function widemul{T, f, U, g}(x::FD{T, f}, y::FD{U, g})
+Base.@pure function widemul(x::FD{T, f}, y::FD{U, g}) where {T, f, U, g}
```
Comment (TotalVerb) [inline src/FixedPointDecimals.jl]: `x::FD{<:Any, f}, y::FD{<:Any, g}` could save two type parameters

File: src/FixedPointDecimals.jl [hunk 3]
```diff
@@ -283,56 +283,59 @@ function convert{T, f, U, g}(::Type{FD{T, f}}, x::FD{U, g})
...
 
-convert{TR <: Rational, T, f}(::Type{TR}, x::FD{T, f})::TR = x.i // coefficient(FD{T, f})
+function convert(::Type{TR}, x::FD{T, f}) where {TR <: Rational, T, f}
+    convert(TR, x.i // coefficient(FD{T, f}))::TR
+end
 
-promote_rule{T, f, TI <: Integer}(::Type{FD{T, f}}, ::Type{TI}) = FD{T, f}
-promote_rule{T, f, TF <: AbstractFloat}(::Type{FD{T, f}}, ::Type{TF}) = TF
-promote_rule{T, f, TR}(::Type{FD{T, f}}, ::Type{Rational{TR}}) = Rational{TR}
+promote_rule(::Type{FD{T, f}}, ::Type{TI}) where {T, f, TI <: Integer} = FD{T, f}
```
Comment (TotalVerb) [inline src/FixedPointDecimals.jl]: `::Type{<:Integer}` and we can drop the TI parameter

File: src/FixedPointDecimals.jl [hunk 4]
```diff
@@ -283,56 +283,59 @@ function convert{T, f, U, g}(::Type{FD{T, f}}, x::FD{U, g})
...
+promote_rule(::Type{FD{T, f}}, ::Type{TF}) where {T, f, TF <: AbstractFloat} = TF
```
Comment (TotalVerb) [inline src/FixedPointDecimals.jl]: `::Type{<:FD}` can probably work to drop the `T` and `f` parameters

File: src/FixedPointDecimals.jl [hunk 5]
```diff
@@ -283,56 +283,59 @@ function convert{T, f, U, g}(::Type{FD{T, f}}, x::FD{U, g})
...
+promote_rule(::Type{FD{T, f}}, ::Type{Rational{TR}}) where {T, f, TR} = Rational{TR}
```
Comment (TotalVerb) [inline src/FixedPointDecimals.jl]: `::Type{<:FD}`

**Comment by TotalVerb** [comment]
Looks good, except please update `.travis.yml` and appveyor to drop 0.5 from testing.

File: src/FixedPointDecimals.jl (hunk 1 above)
Comment (omus) [reply src/FixedPointDecimals.jl]: I changed this just to visually match the rest of the code

Comment (omus) [reply src/FixedPointDecimals.jl]: @TotalVerb do you still want to change this?

Comment (TotalVerb) [reply src/FixedPointDecimals.jl]: How about `::Type{<:FD{BigInt}}`, which is shorter? It wouldn't visually match everything else but would be cleaner I think, with fewer `where`s.

Comment (omus) [reply src/FixedPointDecimals.jl]: I like it. I ended up with:
```julia
floattype(::Type{<:FD{T}}) where {T<:Union{Int8, UInt8, Int16, UInt16}} = Float32
floattype(::Type{<:FD{T}}) where {T<:Integer} = Float64
floattype(::Type{<:FD{BigInt}}) = BigFloat
```

**Pull request merged**